File: java/src/java/com/d_project/qrcode/QR8BitByte.java
package com.d_project.qrcode;

import java.io.UnsupportedEncodingException;

/**
 * QR8BitByte
 * @author Kazuhiko Arase 
 */
class QR8BitByte extends QRData {
	
	public QR8BitByte(String data) {
		super(Mode.MODE_8BIT_BYTE, data);
	}
	
	public void write(BitBuffer buffer) {

		try {

			byte[] data = getData().getBytes(QRUtil.getJISEncoding() );

			for (int i = 0; i < data.length; i++) {
				buffer.put(data[i], 8);
			}
			
		} catch(UnsupportedEncodingException e) {
			throw new RuntimeException(e.getMessage() );
		}		
	}
	
	public int getLength() {
		try {
			return getData().getBytes(QRUtil.getJISEncoding() ).length;
		} catch(UnsupportedEncodingException e) {
			throw new RuntimeException(e.getMessage() );
		}
	}
	
}

File: ts/src/ts/com/d_project/qrcode/QR8BitByte.ts
import QRData from './QRData';
import Mode from './Mode';
import BitBuffer from './BitBuffer';
import QRCode from './QRCode';

/**
 * QR8BitByte
 * @author Kazuhiko Arase
 */
export default class extends QRData {

    constructor(data: string) {
        super(Mode.MODE_8BIT_BYTE, data);
    }

    public write(buffer: BitBuffer): void {
        var data = QRCode.stringToBytes(this.getData());
        for (var i = 0; i < data.length; i += 1) {
            buffer.put(data[i], 8);
        }
    }

    public getLength(): number {
        return QRCode.stringToBytes(this.getData()).length;
    }
}
